roles/kubernetes/master/tasks/main.yml

```diff
@@ -87,8 +87,21 @@
     src: manifests/kube-podmaster.manifest.j2
     dest: "{{ kube_manifest_dir }}/kube-podmaster.manifest"
 
-- name: restart kubelet
-  service:
-    name: kubelet
-    state: restarted
-  changed_when: false
+# Run daemonsets
+- name: Start Resources
+  kube:
+    name: "{{ item.item.name }}"
+    namespace: kube-system
+    kubectl: "{{bin_dir}}/kubectl"
+    resource: "{{item.item.type}}"
+    filename: /etc/kubernetes/{{item.item.file}}
+    state: "{{item.changed | ternary('latest','present') }}"
+  with_items: "{{proxy_ds_manifest.results | default([]) | union(dnsmasq_manifests.results|default([]))}}"
+  when: inventory_hostname == groups['kube-master'][0]
+
+- name: Check for dnsmasq port (pulling image and running container)
+  wait_for:
```

Comment:
starting dnsmasq here make it requierment. It should be easy to remove the role dnsmasq and trplace it by something else. 
by spliting some tasks of the role here break this separation